auth/src/main/java/com/firebase/ui/auth/ui/idp/AuthMethodPickerActivity.java

```diff
@@ -127,6 +126,10 @@ private void populateIdpList(List<IdpConfig> providers) {
                     loginButton = getLayoutInflater()
                             .inflate(R.layout.idp_button_twitter, btnHolder, false);
                     break;
+                case EmailAuthProvider.PROVIDER_ID:
+                    loginButton = getLayoutInflater()
+                            .inflate(R.layout.provider_button_email, btnHolder, false);
```

Comment:
This is where things would have gotten messy without the `Provider` refactor: I would have had to keep track of the email provider index and somehow insert it somewhere inside this for loop without it actually being part of the `mProviders` list. It would have been possible, but very messy.